Sources/BraintreePayPalNativeCheckout/BTPayPalNativeError.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+
+/// Error returned from the native PayPal flow
+enum BTPayPalNativeError: Error, CustomNSError, LocalizedError, Equatable  {
+    static func == (lhs: BTPayPalNativeError, rhs: BTPayPalNativeError) -> Bool {
+        lhs.errorCode == rhs.errorCode
+    }
+
+    /// Request is not of type BTPayPalNativeCheckoutRequest or BTPayPalNativeVaultRequest
+    case invalidRequest
+
+    /// Failed to fetch Braintree configuration
+    case fetchConfigurationFailed
+
+    /// PayPal is not enabled for this merchant in the Braintree Control Panel
+    case payPalNotEnabled
+
+    /// Could not find PayPal client ID in the Braintree configuration
+    case payPalClientIDNotFound
+
+    /// Invalid environment identifier found in the Braintree configuration
+    case invalidEnvironment
+
+    /// Failed to create PayPal order
+    case orderCreationFailed(Error)
+
+    /// PayPal flow was canceled by the user
+    case canceled
+
+    /// PayPalCheckout SDK returned an error
+    case checkoutSDKFailed
+
+    /// Tokenization with the Braintree Gateway failed
+    case tokenizationFailed(Error)
+
+    /// Failed to parse tokenization result
+    case parsingTokenizationResultFailed
+
+    case invalidJSONResponse
+
+    static var errorDomain: String {
+        "com.braintreepayments.BTPaypalNativeCheckoutErrorDomain"
+    }
+
+    var errorCode: Int {
```

Comment:
Up to you, but I was on board with having `BTPayPalNativeError` conform to `Int` and using `rawValue` here. I may have worded liking that approach poorly so sorry if it caused confusion!

Reply:
Changed it to this approach so we can encapsulate other errors (`orderCreationFailed(Error)` and `tokenizationFailed(Error)`) to better handle errors thrown by the server response